BlazorMarkDownAppJwt/BlazorMarkDownAppJwt/Server/Controllers/MarkDownController.cs

```diff
@@ -0,0 +1,176 @@
+﻿using BlazorMarkDownAppJwt.Server.Services.MarkDowns;
+using BlazorMarkDownAppJwt.Shared;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Authorization;
+using System.Net;
+
+namespace BlazorMarkDownAppJwt.Server.Controllers
+{
+    [ApiController]
+    public class MarkDownController : Controller
+    {
+        private IDocumentService markDownService { get; }
+
+        public MarkDownController(IDocumentService markDownService)
+        {
+            this.markDownService = markDownService;
+        }
+
+        [HttpGet]
+        [Route("api/markdown")]
+        [ProducesResponseType(StatusCodes.Status200OK, Type = typeof(MarkDownModel))]
+        public async Task<IActionResult> Get([FromQuery] long currentId)
+        {
+            try
+            {
+
+                var markDownModel = new MarkDownModel
+                {
+                    Body = string.Empty,
+                };
+
+                var document = await markDownService.GetDocument(currentId);
+
+                if (!string.IsNullOrWhiteSpace(document?.MarkDown))
+                {
+
+                    markDownModel.Body = document.MarkDown;
+                    markDownModel.Id = document.Id;
+                }
+
+                return Ok(markDownModel);
+            }
+            catch (Exception ex)
+            {
+                return StatusCode((int)HttpStatusCode.InternalServerError, ex);
+            }
+
+        }
+
+        [HttpGet]
+        [Route("api/markdown/readme/")]
+        [ProducesResponseType(StatusCodes.Status200OK, Type = typeof(MarkDownModel))]
+        public async Task<IActionResult> GetReadMe()
+        {
+            try
+            {
+
+                var markDownModel = new MarkDownModel
+                {
+                    Body = string.Empty,
+                };
+
+                var document = await markDownService.GetReadMeDocument();
+
+                if (document?.MarkDown != null)
+                {
+
+                    markDownModel.Body = document.MarkDown;
+                }
+
+                return Ok(markDownModel);
+            }
+            catch (Exception ex)
+            {
+                return StatusCode((int)HttpStatusCode.InternalServerError, ex);
+            }
+
+        }
+
+        [HttpGet]
+        [Route("api/markdowns/")]
+        [ProducesResponseType(StatusCodes.Status200OK, Type = typeof(List<MarkDownModel>))]
+        public async Task<IActionResult> GetAll()

```

Comment:
**Remark 1&&2**

Reply:
**Remark 5** 
I advice you to use as result type : `Task<List<MarkDownModel>>`  or `Task<ActionResult<MarkDownModel>> `. It will allow ASP.NET Core and swagger to map correctly your result to the good type without to maintain it with custom attribute

Reply:
I could resolve all Remark 1&&2&&5 on the commit : 
https://github.com/clemalex06/BlazorMarkDown/pull/8/commits/614a92a5082bffb7dba45a98b6d1eef6dfdb552e